src/Sylius/Component/Product/Factory/ProductFactory.php

```diff
@@ -72,6 +78,26 @@ public function createNew()
     /**
      * {@inheritdoc}
      */
+    public function createTyped($type)
+    {
+        Assert::oneOf($type, [ProductInterface::SIMPLE, ProductInterface::CONFIGURABLE]);
+
+        if (ProductInterface::SIMPLE === $type) {
+            $product = $this->createNew();
```

Comment:
Not really. In this line I have used a method defined in this class(which provides me new product with master variant defined). [This call](https://github.com/Sylius/Sylius/pull/4947/files#diff-d6e4bdac05336ee91ff1323b45903734R92) will create a pure Product instance without master variant defined.